res/enkf/state_map.py

```diff
@@ -1,137 +1,54 @@
-
-        if not isinstance(index, int):
-            raise TypeError("Expected an integer")
-
-        if not isinstance(value, RealizationStateEnum):
-            raise TypeError("Expected a RealizationStateEnum")
-
-        if index < 0:
-            index += len(self)
-        if index < 0:
-            raise IndexError(f"Index out of range: {index} < 0")
-
-        self._iset(index, value)
-
-    @classmethod
-    def isLegalTransition(cls, realization_state1, realization_state2):
-        """@rtype: bool"""
-
-        if not isinstance(realization_state1, RealizationStateEnum) or not isinstance(
-            realization_state2, RealizationStateEnum
-        ):
-            raise TypeError("Expected a RealizationStateEnum")
-
-        return cls._is_legal_transition(realization_state1, realization_state2)
-
-    def isReadOnly(self):
-        """@rtype: bool"""
-        return self._is_read_only()
-
-    def selectMatching(self, select_mask) -> List[bool]:
-        assert isinstance(select_mask, RealizationStateEnum)
-        return list(_lib.state_map.select_matching(self, select_mask.value, True))
-
-    def deselectMatching(self, select_mask: RealizationStateEnum) -> List[bool]:
-        assert isinstance(select_mask, RealizationStateEnum)
-        return list(_lib.state_map.select_matching(self, select_mask.value, False))
-
-    def realizationList(self, state_value: RealizationStateEnum) -> List[int]:
-        """Will create an integer list of all realisations with state equal to
-        state_value."""
-        mask = self.createMask(state_value)
-        return [idx for idx, value in enumerate(mask) if value]
-
-    def createMask(self, state_value: RealizationStateEnum) -> List[bool]:
-        """Will create a bool list of all realisations with state equal to
-        state_value."""
-        return self.selectMatching(state_value)
-
-    def free(self):
-        self._free()
-
-    def __repr__(self):
-        ro = "read only" if self.isReadOnly() else "read/write"
-        return f"StateMap(size = {len(self)}, {ro}) {self._ad_str()}"
-
-    def load(self, filename):
-        if not self._fread(filename):
-            raise IOError(f"Failed to load state map from:{filename}")
-
-    def save(self, filename):
-        self._fwrite(filename)
+def __getitem__(self, iens: int) -> RealizationStateEnum:
+    if iens < len(self):
+        return self._get(iens)
+    raise IndexError(f"StateMap index out of range: {iens} not in [0, {len(self)})")
+
+
+def __iter__(self):
+    index = 0
+    size = len(self)
+
+    while index < size:
+        yield self[index]
+        index += 1
+
+
+def realizationList(self, state_value):
+    """
+    Will create a list of all realisations with state equal to state_value.
+
+    @type state_value: RealizationStateEnum
+    @rtype: ecl.util.IntVector
+    """
+    return [index for index, value in enumerate(self) if value == state_value]
+
+
+def createMask(self, state_value):
```

Comment:
The old behavior returned a list of bools not a "BoolVector"

Reply:
could we keep the typing

Reply:
oki